packages/manager/modules/vps/src/dashboard/tile/configuration/upgrade/routing.js

```diff
@@ -0,0 +1,222 @@
+
+import component from './component';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('vps.detail.dashboard.configuration.upgrade', {
+    url: '/upgrade/{upgradeType:memory|storage}?upgradeStatus&upgradeOrderId',
+    layout: 'ouiModal',
+    redirectTo: (transition) => {
+      const $q = transition.injector().get('$q');
+      const upgradeTypePromise = transition.injector().getAsync('upgradeType');
+      const configurationTilePromise = transition
+        .injector()
+        .getAsync('configurationTile');
+
+      return $q
+        .all({
+          upgradeType: upgradeTypePromise,
+          configurationTile: configurationTilePromise,
+        })
+        .then(({ upgradeType, configurationTile }) => {
+          const upgradeToPlan = get(
+            configurationTile,
+            `upgrades[${upgradeType}].plan`,
+          );
+
+          // check if the upgrade is available
+          if (!upgradeToPlan) {
+            return 'vps.detail.dashboard';
+          }
+
+          const upgradeModelPlan = get(
+            configurationTile,
+            `model[${upgradeType}]`,
+          );
+
+          // check if the model is setted correctly => if we access to the state
+          // directly (without clicking on oui-radio in dashboard state)
+          if (upgradeModelPlan.planCode !== upgradeToPlan.planCode) {
+            set(configurationTile.model, upgradeType, upgradeToPlan);
+          }
+
+          return null;
+        });
+    },
+    onExit: (transition) => {
+      const configurationTilePromise = transition
+        .injector()
+        .getAsync('configurationTile');
+
+      return configurationTilePromise.then((configurationTile) => {
+        set(
+          configurationTile.model,
+          'memory',
+          VpsConfigurationTile.currentPlan,
+        );
+        set(
+          configurationTile.model,
+          'storage',
+          VpsConfigurationTile.currentPlan,
+        );
+      });
+    },
+    component: component.name,
+    resolve: {
+      upgradeType: /* @ngInject */ ($transition$) =>
+        $transition$.params().upgradeType,
+
+      upgradeStatus: /* @ngInject */ ($transition$) =>
+        $transition$.params().upgradeStatus,
+
+      upgradeOrderId: /* @ngInject */ ($transition$) =>
+        $transition$.params().upgradeOrderId,
+
+      redirectTo: () => 'vps.detail.dashboard',
+
+      upgradeInfo: /* @ngInject */ (
+        configurationTile,
+        serviceName,
+        upgradeStatus,
+        upgradeType,
+        vpsUpgrade,
+      ) => {
+        if (upgradeStatus === 'success') {
+          return true;
+        }
+
+        return vpsUpgrade.getUpgrade(
+          serviceName,
+          get(configurationTile.upgrades, `${upgradeType}.plan.planCode`),
+          { quantity: 1 },
+        );
+      },
+
+      hasDefaultPaymentMethod: /* @ngInject */ (ovhPaymentMethod) =>
+        ovhPaymentMethod.hasDefaultPaymentMethod(),
```

Comment:
This resolve is available here: https://github.com/ovh/manager/blob/feat/vps-new-range-upgrade/packages/manager/modules/vps/src/routing.js#L36

Reply:
We need first to have your code merged in order for me to use this resolve.

Reply:
Yes sorry, forgot to write it as a reminder ;)